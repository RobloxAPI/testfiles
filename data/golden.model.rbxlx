<roblox version="4">
	<Meta name="MetaKeyOverride">MetaValueA</Meta>
	<Meta name="MetaKeyOverride">MetaValueB</Meta>
	<Item class="TestValues">
		<Properties>
			<bool name="BoolTrue">true</bool>
			<bool name="BoolFalse">false</bool>
			<float name="FloatGravity">196.2</float>
			<float name="FloatRound">99003.78125</float>
			<float name="FloatFive">-0.55555555555555555</float>
			<float name="FloatPi">3.14159274</float>
			<float name="FloatE">2.71828182845904523536028747135266249775724709369995957496696763</float>
			<float name="FloatPi">3.14159265358979323846264338327950288419716939937510582097494459</float>
			<float name="FloatPhi">1.61803398874989484820458683436563811772030917980576286213544862</float>
			<float name="FloatSqrt2">1.41421356237309504880168872420969807856967187537694807317667974</float>
			<float name="FloatSqrtE">1.64872127070012814684865078781416357165377610071014801157507931</float>
			<float name="FloatSqrtPi">1.77245385090551602729816748334114518279754945612238712821380779</float>
			<float name="FloatSqrtPhi">1.27201964951406896425242246173749149171560804184009624861664038</float>
			<float name="FloatLn2">0.693147180559945309417232121458176568075500134360255254120680009</float>
			<float name="FloatLn10">2.30258509299404568401799145468436420760110148862877297603332790</float>
			<double name="DoubleGravity">196.2</double>
			<double name="DoubleRound">99003.78125</double>
			<double name="DoubleFive">-0.55555555555555555</double>
			<double name="DoublePi">3.14159274</double>
			<double name="DoubleE">2.71828182845904523536028747135266249775724709369995957496696763</double>
			<double name="DoublePi">3.14159265358979323846264338327950288419716939937510582097494459</double>
			<double name="DoublePhi">1.61803398874989484820458683436563811772030917980576286213544862</double>
			<double name="DoubleSqrt2">1.41421356237309504880168872420969807856967187537694807317667974</double>
			<double name="DoubleSqrtE">1.64872127070012814684865078781416357165377610071014801157507931</double>
			<double name="DoubleSqrtPi">1.77245385090551602729816748334114518279754945612238712821380779</double>
			<double name="DoubleSqrtPhi">1.27201964951406896425242246173749149171560804184009624861664038</double>
			<double name="DoubleLn2">0.693147180559945309417232121458176568075500134360255254120680009</double>
			<double name="DoubleLn10">2.30258509299404568401799145468436420760110148862877297603332790</double>
			<int name="IntMinUnderflow">-2147483649</int>
			<int name="IntMin">-2147483648</int>
			<int name="IntMinIn">-2147483647</int>
			<int name="IntMinusTwo">-2</int>
			<int name="IntMinusOne">-1</int>
			<int name="IntZero">0</int>
			<int name="IntOne">1</int>
			<int name="IntMaxIn">2147483646</int>
			<int name="IntMax">2147483647</int>
			<int name="IntMaxOverflow">2147483648</int>
			<int64 name="Int64MinUnderflow">-9223372036854775809</int64>
			<int64 name="Int64Min">-9223372036854775808</int64>
			<int64 name="Int64MinIn">-9223372036854775807</int64>
			<int64 name="Int64MinusTwo">-2</int64>
			<int64 name="Int64MinusOne">-1</int64>
			<int64 name="Int64Zero">0</int64>
			<int64 name="Int64One">1</int64>
			<int64 name="Int64MaxIn">9223372036854775806</int64>
			<int64 name="Int64Max">9223372036854775807</int64>
			<int64 name="Int64MaxOverflow">9223372036854775808</int64>
		</Properties>
	</Item>
	<Item class="ReferentA" referent="aaa"></Item>
	<Meta name="MetaKeyMiddle">MetaValueMiddle</Meta>
	<Meta name="MetaASort">MetaValueSort</Meta>
	<Item class="RefererA">
		<Properties>
			<Ref name="ReferenceA">aaa</Ref>
		</Properties>
	</Item>
	<Item class="RefererB">
		<Properties>
			<Ref name="ReferenceB">bbb</Ref>
		</Properties>
	</Item>
	<Item class="ReferentB" referent="bbb"></Item>
</roblox>
