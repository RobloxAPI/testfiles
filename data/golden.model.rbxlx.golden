Format: rbxlx
Struct: model
Data
	Metadata
		0
			Key: "MetaASort"
			Value: "MetaValueSort"
		1
			Key: "MetaKeyMiddle"
			Value: "MetaValueMiddle"
		2
			Key: "MetaKeyOverride"
			Value: "MetaValueB"
	Instances
		0
			ClassName: "TestValues"
			IsService: false
			Reference: 0
			Properties
				0
					Name: "Axes"
					Type: "Axes"
					Value
						X: false
						Y: true
						Z: true
				1
					Name: "BoolFalse"
					Type: "Bool"
					Value: false
				2
					Name: "BoolTrue"
					Type: "Bool"
					Value: true
				3
					Name: "BrickColor"
					Type: "BrickColor"
					Value: 1234567890
				4
					Name: "CFrame"
					Type: "CFrame"
					Value
						Position
							X: 99000.7812
							Y: -99001.7812
							Z: 99002.7812
						Rotation
							R00: -99003.7812
							R01: 99004.7812
							R02: -99005.7812
							R10: 99006.7812
							R11: -99007.7812
							R12: 99008.7812
							R20: -99009.7812
							R21: 99010.7812
							R22: -99011.7812
				5
					Name: "Color3"
					Type: "Color3"
					Value
						R: 99000.7812
						G: -99001.7812
						B: 99002.7812
				6
					Name: "Color3uint8"
					Type: "Color3uint8"
					Value
						R: 210
						G: 45
						B: 212
				7
					Name: "ColorSequence"
					Type: "ColorSequence"
					Value
						0
							Time: 99000.7812
							Value
								R: -99001.7812
								G: 99002.7812
								B: -99003.7812
							Envelope: 99004.7812
						1
							Time: -99005.7812
							Value
								R: 99006.7812
								G: -99007.7812
								B: 99008.7812
							Envelope: -99009.7812
						2
							Time: 99010.7812
							Value
								R: -99011.7812
								G: 99012.7812
								B: -99013.7812
							Envelope: 99014.7812
				8
					Name: "ContentBytes"
					Type: "Content"
					Value
						Length: 23
						| 48 65 6c 6c 6f 00 01 02  09 0a 7e 7f 80 81 fd fe |Hello.....~.....|
						| ff 77 6f 72 6c 64 21                             |.world!|
				9
					Name: "ContentEscaped"
					Type: "Content"
					Value: "Hello\n\tworld!\n"
				10
					Name: "ContentNull"
					Type: "Content"
					Value: ""
				11
					Name: "ContentSimple"
					Type: "Content"
					Value: "Hello world!"
				12
					Name: "ContentUnicode"
					Type: "Content"
					Value: "Hello, 世界!"
				13
					Name: "DoubleE"
					Type: "Double"
					Value: 2.7182818284590451
				14
					Name: "DoubleFive"
					Type: "Double"
					Value: -0.55555555555555558
				15
					Name: "DoubleGravity"
					Type: "Double"
					Value: 196.19999999999999
				16
					Name: "DoubleLn10"
					Type: "Double"
					Value: 2.3025850929940459
				17
					Name: "DoubleLn2"
					Type: "Double"
					Value: 0.69314718055994529
				18
					Name: "DoublePhi"
					Type: "Double"
					Value: 1.6180339887498949
				19
					Name: "DoublePi"
					Type: "Double"
					Value: 3.1415926535897931
				20
					Name: "DoubleRound"
					Type: "Double"
					Value: 99003.78125
				21
					Name: "DoubleSqrt2"
					Type: "Double"
					Value: 1.4142135623730951
				22
					Name: "DoubleSqrtE"
					Type: "Double"
					Value: 1.6487212707001282
				23
					Name: "DoubleSqrtPhi"
					Type: "Double"
					Value: 1.272019649514069
				24
					Name: "DoubleSqrtPi"
					Type: "Double"
					Value: 1.7724538509055161
				25
					Name: "Faces"
					Type: "Faces"
					Value
						Right: true
						Top: false
						Back: true
						Left: false
						Bottom: true
						Front: false
				26
					Name: "FloatE"
					Type: "Float"
					Value: 2.71828175
				27
					Name: "FloatFive"
					Type: "Float"
					Value: -0.555555582
				28
					Name: "FloatGravity"
					Type: "Float"
					Value: 196.199997
				29
					Name: "FloatLn10"
					Type: "Float"
					Value: 2.30258512
				30
					Name: "FloatLn2"
					Type: "Float"
					Value: 0.693147182
				31
					Name: "FloatPhi"
					Type: "Float"
					Value: 1.61803401
				32
					Name: "FloatPi"
					Type: "Float"
					Value: 3.14159274
				33
					Name: "FloatRound"
					Type: "Float"
					Value: 99003.7812
				34
					Name: "FloatSqrt2"
					Type: "Float"
					Value: 1.41421354
				35
					Name: "FloatSqrtE"
					Type: "Float"
					Value: 1.64872122
				36
					Name: "FloatSqrtPhi"
					Type: "Float"
					Value: 1.27201962
				37
					Name: "FloatSqrtPi"
					Type: "Float"
					Value: 1.7724539
				38
					Name: "Int64Max"
					Type: "Int64"
					Value: 9223372036854775807
				39
					Name: "Int64MaxIn"
					Type: "Int64"
					Value: 9223372036854775806
				40
					Name: "Int64MaxOverflow"
					Type: "Int64"
					Value: 9223372036854775807
				41
					Name: "Int64MaxOverflowMore"
					Type: "Int64"
					Value: 9223372036854775807
				42
					Name: "Int64Min"
					Type: "Int64"
					Value: -9223372036854775808
				43
					Name: "Int64MinIn"
					Type: "Int64"
					Value: -9223372036854775807
				44
					Name: "Int64MinUnderflow"
					Type: "Int64"
					Value: -9223372036854775808
				45
					Name: "Int64MinUnderflowMore"
					Type: "Int64"
					Value: -9223372036854775808
				46
					Name: "Int64MinusOne"
					Type: "Int64"
					Value: -1
				47
					Name: "Int64MinusTwo"
					Type: "Int64"
					Value: -2
				48
					Name: "Int64One"
					Type: "Int64"
					Value: 1
				49
					Name: "Int64Zero"
					Type: "Int64"
					Value: 0
				50
					Name: "IntMax"
					Type: "Int"
					Value: 2147483647
				51
					Name: "IntMaxIn"
					Type: "Int"
					Value: 2147483646
				52
					Name: "IntMaxOverflow"
					Type: "Int"
					Value: 2147483647
				53
					Name: "IntMaxOverflowMore"
					Type: "Int"
					Value: 2147483647
				54
					Name: "IntMin"
					Type: "Int"
					Value: -2147483648
				55
					Name: "IntMinIn"
					Type: "Int"
					Value: -2147483647
				56
					Name: "IntMinUnderflow"
					Type: "Int"
					Value: -2147483648
				57
					Name: "IntMinUnderflowMore"
					Type: "Int"
					Value: -2147483648
				58
					Name: "IntMinusOne"
					Type: "Int"
					Value: -1
				59
					Name: "IntMinusTwo"
					Type: "Int"
					Value: -2
				60
					Name: "IntOne"
					Type: "Int"
					Value: 1
				61
					Name: "IntZero"
					Type: "Int"
					Value: 0
				62
					Name: "NumberRange"
					Type: "NumberRange"
					Value
						Min: 99000.7812
						Max: -99001.7812
				63
					Name: "NumberSequence"
					Type: "NumberSequence"
					Value
						0
							Time: 99000.7812
							Value: -99001.7812
							Envelope: 99002.7812
						1
							Time: -99003.7812
							Value: 99004.7812
							Envelope: -99005.7812
						2
							Time: 99006.7812
							Value: -99007.7812
							Envelope: 99008.7812
						3
							Time: -99009.7812
							Value: 99010.7812
							Envelope: -99011.7812
				64
					Name: "PhysicalPropertiesFalse"
					Type: "PhysicalProperties"
					Value
						CustomPhysics: false
				65
					Name: "PhysicalPropertiesFalseEmpty"
					Type: "PhysicalProperties"
					Value
						CustomPhysics: false
				66
					Name: "PhysicalPropertiesTrue"
					Type: "PhysicalProperties"
					Value
						CustomPhysics: true
						Density: 99000.7812
						Friction: -99001.7812
						Elasticity: 99002.7812
						FrictionWeight: -99003.7812
						ElasticityWeight: 99004.7812
				67
					Name: "PhysicalPropertiesTrueEmpty"
					Type: "PhysicalProperties"
					Value
						CustomPhysics: true
						Density: 0
						Friction: 0
						Elasticity: 0
						FrictionWeight: 0
						ElasticityWeight: 0
				68
					Name: "ProtectedStringBytes"
					Type: "ProtectedString"
					Value
						Length: 23
						| 48 65 6c 6c 6f 00 01 02  09 0a 7e 7f 80 81 fd fe |Hello.....~.....|
						| ff 77 6f 72 6c 64 21                             |.world!|
				69
					Name: "ProtectedStringEscaped"
					Type: "ProtectedString"
					Value: "Hello\n\tworld!\n"
				70
					Name: "ProtectedStringSimple"
					Type: "ProtectedString"
					Value: "Hello world!"
				71
					Name: "ProtectedStringUnicode"
					Type: "ProtectedString"
					Value: "Hello, 世界!"
				72
					Name: "Ray"
					Type: "Ray"
					Value
						Origin
							X: 99000.7812
							Y: -99001.7812
							Z: 99002.7812
						Direction
							X: -99003.7812
							Y: 99004.7812
							Z: -99005.7812
				73
					Name: "Rect2D"
					Type: "Rect2D"
					Value
						Min
							X: 99000.7812
							Y: -99001.7812
						Max
							X: 99002.7812
							Y: -99003.7812
				74
					Name: "SharedStringBytes"
					Type: "SharedString"
					Value
						Length: 23
						| 48 65 6c 6c 6f 00 01 02  09 0a 7e 7f 80 81 fd fe |Hello.....~.....|
						| ff 77 6f 72 6c 64 21                             |.world!|
				75
					Name: "SharedStringEscaped"
					Type: "SharedString"
					Value: "Hello\n\tworld!\n"
				76
					Name: "SharedStringSimple"
					Type: "SharedString"
					Value: "Hello world!"
				77
					Name: "SharedStringUnicode"
					Type: "SharedString"
					Value: "Hello, 世界!"
				78
					Name: "StringBytes"
					Type: "String"
					Value
						Length: 23
						| 48 65 6c 6c 6f 00 01 02  09 0a 7e 7f 80 81 fd fe |Hello.....~.....|
						| ff 77 6f 72 6c 64 21                             |.world!|
				79
					Name: "StringEscaped"
					Type: "String"
					Value: "Hello\n\tworld!\n"
				80
					Name: "StringSimple"
					Type: "String"
					Value: "Hello world!"
				81
					Name: "StringUnicode"
					Type: "String"
					Value: "Hello, 世界!"
				82
					Name: "Token"
					Type: "Token"
					Value: 1234567890
				83
					Name: "UDim"
					Type: "UDim"
					Value
						Scale: 99000.7812
						Offset: 1234567890
				84
					Name: "UDim2"
					Type: "UDim2"
					Value
						X
							Scale: 99000.7812
							Offset: 1234567890
						Y
							Scale: -99001.7812
							Offset: -1234567891
				85
					Name: "Vector2"
					Type: "Vector2"
					Value
						X: 99000.7812
						Y: -99001.7812
				86
					Name: "Vector3"
					Type: "Vector3"
					Value
						X: 99000.7812
						Y: -99001.7812
						Z: 99002.7812
			Children
		1
			ClassName: "ReferentA"
			IsService: false
			Reference: 1
			Properties
			Children
		2
			ClassName: "RefererA"
			IsService: false
			Reference: 2
			Properties
				0
					Name: "ReferenceA"
					Type: "Reference"
					Value: 1
			Children
		3
			ClassName: "RefererB"
			IsService: false
			Reference: 3
			Properties
				0
					Name: "ReferenceB"
					Type: "Reference"
					Value: 4
			Children
		4
			ClassName: "ReferentB"
			IsService: false
			Reference: 4
			Properties
			Children
